fs/spiffs/src/spiffs_vfs.c

```diff
@@ -330,7 +330,14 @@ static int spiffs_readdir_callback(FAR struct spiffs_s *fs,
       DEBUGASSERT(dir != NULL);
       entryp = &dir->fd_dir;
 
-      strncpy(entryp->d_name, (FAR char *)objhdr.name, NAME_MAX + 1);
+      /* Skip the leading '/'. */
+
+      if (objhdr.name[0] != '/')
```

Comment:
how about:
if (objhdr.name[0] != '/')
  {
    strncpy(entryp->d_name, (FAR char *)objhdr.name , NAME_MAX + 1);
  }
else
  {
    strncpy(entryp->d_name, (FAR char *)objhdr.name + 1, NAME_MAX + 1);
  }

Reply:
do you have specific reasons to make this that smart?
generally i don't think it's a good idea to put this kind of smart things unless absolutely necessary.

Reply:
Sure, it's fine if you confirm the lead '/' always exist.